compaction.js

```diff
@@ -25,23 +25,25 @@ const NO_TRUNCATE = 0xffffffff
  *
  * - bytes 0..3: UInt32LE for the version of this file format
  *               smallest version is 1.
- * - bytes 4..7: UInt32LE for block index where to perform truncation
- *               where 0xFFFFFFFF means no truncation to-be-done yet
- * - bytes 8..11: UInt32LE for the blockIndex to-be-compacted
- * - bytes 12..15: UInt32LE for the 1st unshifted record's offset
- * - bytes 16..(16+blockSize-1): blockBuf containing the 1st unshifted record
+ * - bytes 4..7: UInt32LE for the startOffset, usually the start of some block
+ * - bytes 8..11: UInt32LE for block index where to perform truncation
+ *                where 0xFFFFFFFF means no truncation to-be-done yet
+ * - bytes 12..15: UInt32LE for the blockIndex to-be-compacted
+ * - bytes 16..19: UInt32LE for the 1st unshifted record's offset
+ * - bytes 20..(20+blockSize-1): blockBuf containing the 1st unshifted record
  */
 function PersistentState(logFilename, blockSize) {
   const raf = RAF(getStateFilename(logFilename))
   const writeLock = mutexify()
-  const stateFileSize = 4 + 4 + 4 + 4 + blockSize
+  const stateFileSize = 4 + 4 + 4 + 4 + 4 + blockSize
 
   function load(cb) {
     raf.stat(function onRAFStatDone(err, stat) {
       const fileSize = !err && stat ? stat.size : -1
       if (fileSize <= 0) {
         const state = {
           version: 1,
```

Comment:
Technically we should bump the version number (and add utils that convert from the old version to the new version) but given that this is not in production yet, it doesn't hurt to keep it the same.